src/core/router/net_db/impl.cc

```diff
@@ -376,39 +379,39 @@ void NetDb::SaveUpdated() {
       it.second->DeleteBuffer();
       count++;
     } else {
-      // RouterInfo expires after 1 hour if it uses an introducer
+      // RouterInfo expires after N minutes if it uses an introducer
       if (it.second->UsesIntroducer() && ts > it.second->GetTimestamp() +
-          static_cast<std::uint64_t>(NetDbTime::RouterExpiration)) {  // 1 hour
+          static_cast<std::uint32_t>(NetDbTime::RouterExpiration)) {
         it.second->SetUnreachable(true);
-        /// @brief if router count is greater than the threshold check and the router
-        ///  is no longer starting up then continue to check for unreachable routers
+        // if the router count is greater than the threshold check, and the router
+        // is no longer starting up, then continue to check for unreachable routers
       } else if (total >
           static_cast<std::uint16_t>(NetDbSize::RouterCheckUnreachableThreshold)
-          && ts > kovri::context.GetStartupTime() +
-          static_cast<std::uint64_t>(NetDbTime::RouterStartupPeriod)) {
+          && ts > (kovri::context.GetStartupTime() +
+          static_cast<std::uint32_t>(NetDbTime::RouterStartupPeriod)) * 1000) {
         if (kovri::context.IsFloodfill()) {
           if (ts > it.second->GetTimestamp() +
-              static_cast<std::uint64_t>(NetDbTime::RouterExpiration)) {
+              static_cast<std::uint32_t>(NetDbTime::RouterExpiration)) {
             it.second->SetUnreachable(true);
             total--;
           }
-          /// @brief if router count is higher, expiration date for unreachable
-          ///  peers is shorter
+          //  if router count is higher, expiration date for unreachable
+          //  peers is shorter
         } else if (total >
             static_cast<std::uint16_t>(NetDbSize::MaxRouterCheckUnreachable)) {
-          // 30 hours
           if (ts > it.second->GetTimestamp() +
-              30 * static_cast<std::uint64_t>(NetDbTime::RouterExpiration)) {
+              (static_cast<std::uint32_t>(NetDbTime::RouterMinGracePeriod) *
+              static_cast<std::uint32_t>(NetDbTime::RouterExpiration))) {
```

Comment:
As noted in https://github.com/monero-project/kovri/pull/444#discussion_r87044802, here's an example

```cpp
if (ts > it.second->GetTimestamp()
    + static_cast<std::uint32_t>(NetDbTime::RouterMinGracePeriod)
    * static_cast<std::uint32_t>(NetDbTime::RouterExpiration)) {
```

This also applies elsewhere and helps with legibility.